src/hackerspace_online/management/commands/full_clean.py

```diff
@@ -0,0 +1,58 @@
+from django.apps import apps
+from django.core.exceptions import ValidationError
+from django.core.management.base import BaseCommand
+from django_tenants.utils import schema_context
+from tenant.models import Tenant
+
+
+class Command(BaseCommand):
+    LOCAL_APPS = (
+        'quest_manager',
+        'profile_manager',
+        'announcements',
+        'comments',
+        'notifications',
+        'courses',
+        'prerequisites',
+        'badges',
+        'djcytoscape',
+        'portfolios',
+        'utilities',
+        'siteconfig',
+        'tags',
+        'library',
+    )
+    help = "Loops through a list of tenants (all by default), and validates each object in the database using full_clean()"
+
+    def add_arguments(self, parser):
+
+        # optional arguments
+        parser.add_argument(
+            '--tenants', action='store', nargs='+',
+            help='A space separated list of tenant schemas. Specifies which tenant to loop through. Defaults to all tenants'
+        )
+
+    def handle(self, *args, **options):
+        tenant_names = options.get('tenants')
+
+        tenants = Tenant.objects.filter(schema_name__in=tenant_names) if tenant_names else Tenant.objects.all()
+
+        # querying models from these gives a relation error
+        tenants = tenants.exclude(schema_name__in=['library', 'public'])
```

Comment:
Why does library give an relation error?  I understand skipping public.  But why Library?  please add more info to this code comment

Reply:
just tested it and found it actually does not. removed